lua/controller.lua

```diff
@@ -253,49 +258,68 @@ local function controller_get_drawer_index(pos, itemstring)
 	return drawer_net_index
 end
 
+local function controller_insert_to_empty_drawer(pos, stack)
+	local name = stack:get_name()
+	local drawer_net_index = controller_get_drawer_index(pos, name)
+
+	if not drawer_net_index["empty"] then return stack end
+
+	local drawer_pos = drawer_net_index["empty"]["drawer_pos"]
+	local visualid = drawer_net_index["empty"]["visualid"]
+	local content = drawers.drawer_get_content(drawer_pos, visualid)
+
+	-- If the drawer is still empty and the drawer entity is loaded, we will
+	-- put the items in the drawer
+	if content.name == "" and drawers.drawer_visuals[core.serialize(drawer_pos)] then
```

Comment:
`minetest.hash_node_position(pos)` would be more appropriate, and quite a bit faster than serializing.

Reply:
thanks for the hint

Reply:
there are many such improvements possible. There is even a custom function written to do what math.abs() does.

Reply:
:laughing: Oh no. I see there's some more potential, though being out of scope for this PR.

Reply:
actually, until I have a rewrite branch up, this is a good place to tell me what can be improved on the mod